ext/ffi/lib.rs

```diff
@@ -756,44 +558,421 @@ fn op_ffi_get_static(
+    }
+    NativeType::I8 => {
+      let value = unsafe { symbol.cif.call::<i8>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::U16 => {
+      let value = unsafe { symbol.cif.call::<u16>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I16 => {
+      let value = unsafe { symbol.cif.call::<i16>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::U32 => {
+      let value = unsafe { symbol.cif.call::<u32>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I32 => {
+      let value = unsafe { symbol.cif.call::<i32>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::U64 => {
+      let value = unsafe { symbol.cif.call::<u64>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I64 => {
+      let value = unsafe { symbol.cif.call::<i64>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::USize => {
+      let value = unsafe { symbol.cif.call::<usize>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::ISize => {
+      let value = unsafe { symbol.cif.call::<isize>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::F32 => {
+      let value = unsafe { symbol.cif.call::<f32>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::F64 => {
+      let value = unsafe { symbol.cif.call::<f64>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::Pointer => {
+      let value =
+        unsafe { symbol.cif.call::<*const u8>(symbol.ptr, &call_args) } as u64;
+      let ptr = U32x2::from(value);
+      serde_v8::to_v8(scope, ptr).unwrap()
+    }
+  };
+  rv.set(value);
+}
+
+pub fn op_ffi_call_ptr_nonblocking_fn(
+  scope: &mut deno_core::v8::HandleScope,
+  args: deno_core::v8::FunctionCallbackArguments,
+  mut rv: deno_core::v8::ReturnValue,
+) {
+  // SAFETY: deno_core guarantees args.data() is a v8 External pointing to an OpCtx for the isolates lifetime
+  let ctx = unsafe {
+    &*(v8::Local::<v8::External>::cast(args.data().unwrap_unchecked()).value()
+      as *const deno_core::_ops::OpCtx)
+  };
+  let op_id = ctx.id;
+  let promise_id = args.get(0).int32_value(scope).unwrap();
+
+  let state = ctx.state.clone();
 
-  ffi_call(args, symbol)
+  // Track async call & get copy of get_error_class_fn
+  let get_class = {
+    let state = state.borrow();
+    state.tracker.track_async(op_id);
+    state.get_error_class_fn
+  };
+
+  let pointer: U32x2 = serde_v8::from_v8(scope, args.get(1)).unwrap();
+  let fn_def: ForeignFunction = serde_v8::from_v8(scope, args.get(2)).unwrap();
+  let symbol = get_symbol(pointer.into(), fn_def);
```

Comment:
Definitely not a thing for this PR, but eventually it might be useful to divide the function pointer symbol creation and calling into separate ops, where the `UnsafeFnPointer` constructor would already create the symbol and use garbage collectable resources to bind the symbol's lifetime to the `UnsafeFnPointer` instance, after which calling would be nearly 100% equivalent to a normal symbol call. The only difference would be the resource to symbol getting.